cli/flox-rust-sdk/src/flox.rs

```diff
@@ -192,45 +192,80 @@ pub enum FloxhubError {
     InvalidFloxhubBaseUrl(String, #[source] url::ParseError),
 }
 
-use tempfile::TempDir;
-/// Should only be used in the flox crate
-pub fn test_flox_instance() -> (Flox, TempDir) {
-    use crate::models::environment::{global_manifest_path, init_global_manifest};
-
-    let tempdir_handle = tempfile::tempdir_in(std::env::temp_dir()).unwrap();
-
-    let cache_dir = tempdir_handle.path().join("caches");
-    let data_dir = tempdir_handle.path().join(".local/share/flox");
-    let temp_dir = tempdir_handle.path().join("temp");
-    let config_dir = tempdir_handle.path().join("config");
-
-    std::fs::create_dir_all(&cache_dir).unwrap();
-    std::fs::create_dir_all(&temp_dir).unwrap();
-    std::fs::create_dir_all(&config_dir).unwrap();
-
-    let flox = Flox {
-        system: env!("NIX_TARGET_SYSTEM").to_string(),
-        cache_dir,
-        data_dir,
-        temp_dir,
-        config_dir,
-        access_tokens: Default::default(),
-        netrc_file: Default::default(),
-        uuid: Default::default(),
-        floxhub: Floxhub::new(Url::from_str("https://hub.flox.dev").unwrap(), None).unwrap(),
-        floxhub_token: None,
-    };
-
-    init_global_manifest(&global_manifest_path(&flox)).unwrap();
-
-    (flox, tempdir_handle)
+pub mod test_helpers {
```

Comment:
Suggestion: add a `#[cfg(test)]` so this module isn't part of the compiled binary

Reply:
You should be able to include it from another crate as long as the code that's using it is also under a `#[cfg(test)]`.

Reply:
That doesn't seem to be working, maybe because of https://github.com/rust-lang/cargo/issues/8379?

```
> git diff --no-ext-diff
diff --git a/cli/flox-rust-sdk/src/flox.rs b/cli/flox-rust-sdk/src/flox.rs
index 4e31d870..1177b11e 100644
--- a/cli/flox-rust-sdk/src/flox.rs
+++ b/cli/flox-rust-sdk/src/flox.rs
@@ -268,6 +268,8 @@ pub mod tests {

     use super::*;

+    pub const var: bool = false;
+
     /// A fake FloxHub token
     ///
     /// {
diff --git a/cli/flox-rust-sdk/src/models/environment/core_environment.rs b/cli/flox-rust-sdk/src/models/environment/core_environment.rs
index bf0aac53..1c0b24c5 100644
--- a/cli/flox-rust-sdk/src/models/environment/core_environment.rs
+++ b/cli/flox-rust-sdk/src/models/environment/core_environment.rs
@@ -742,6 +742,7 @@ mod tests {

     use super::*;
     use crate::flox::test_helpers::{flox_instance, flox_instance_with_global_lock};
+    use crate::flox::tests::var;

     /// Create a CoreEnvironment with an empty manifest
     ///
diff --git a/cli/flox/src/commands/init/python.rs b/cli/flox/src/commands/init/python.rs
index 72995980..70f2edbb 100644
--- a/cli/flox/src/commands/init/python.rs
+++ b/cli/flox/src/commands/init/python.rs
@@ -712,6 +712,7 @@ mod tests {
     }

     use flox_rust_sdk::flox::test_helpers::flox_instance_with_global_lock;
+    use flox_rust_sdk::flox::tests::var;
     use pretty_assertions::assert_eq;
     use serial_test::serial;

> pushd cli; cargo test; popd
   Compiling flox-rust-sdk v0.0.0 (/Users/matthew/flox/main/cli/flox-rust-sdk)
   Compiling flox v1.0.1 (/Users/matthew/flox/main/cli/flox)
error[E0432]: unresolved import `flox_rust_sdk::flox::tests`
   --> flox/src/commands/init/python.rs:715:30
    |
715 |     use flox_rust_sdk::flox::tests::var;
    |                              ^^^^^ could not find `tests` in `flox`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `flox` (bin "flox" test) due to previous error
```

Reply:
TIL that you can't share `#[cfg(test)]` code across crate boundaries, wtf